framework/include/base/Assembly.h

```diff
@@ -737,6 +737,10 @@ class Assembly
   /// Temporary work vector to keep from reallocating it
   std::vector<dof_id_type> _temp_dof_indices;
 
+  /// Temporary work data for reinitAtPhysical()
+  std::vector<Point> _temp_reference_points;
+  FEType _temp_fe_type;
```

Comment:
@andrsd this is not another FE, but just a lightweight `FEType` object.  And the only thing it saves is the `FEType()` construction on the fly that used to be there.  

Are you wanting to replace the call to FEInterface::inverse_map() with a direct call on the object along the lines of `holder_fe->inverse_map`?  I tried this and it is really annoying/ugly to do since `inverse_map()` is not defined on `FEBase`, so you have to do a downcast, etc.  This is why the `FEInterface` exists actually.

IMO the code is fine, and can probably go in as-is...